### PR TITLE
[WIP] [CI]: Check that we can assemble and link code snippets in CI

### PR DESCRIPTION
**Pull request by rswarbrick**
This PR will have a couple of extra "dummy" commits come and go to check that it triggers CI properly. The commit message for the first commit (the one that should eventually get merged) is:

**[CI] Assemble OTBN code snippets on OTBN change**

This should spot any complete disasters with the tooling.

### CODE REVIEW

**Comment by rswarbrick** [comment]
Right, I've checked and this seems to work right. It triggers on changes to anything with `/otbn/` in the path and runs the OTBN tooling to assemble and link each file in our code snippets directory. The last CI that ran with a "TMP" commit passed the OTBN smoke test (the cross is because I was impatient and stopped it before it finished).

**Comment by rswarbrick** [comment]
Oh, it seems that GitHub deletes all the history of the PR if you mark a draft as ready for review. So you'll have to believe me that I've checked and it does fail correctly if something wonky appears in the code snippets.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -392,9 +392,13 @@ jobs:
       verilator --version
     displayName: Display environment
   - bash: |
-      export PATH="/opt/buildcache/riscv/bin:$PATH"
+      export PATH="${TOOLCHAIN_PATH}/bin:$PATH"
```
Comment (imphil) [inline azure-pipelines.yml:395]:
```suggestion
      export PATH="$TOOLCHAIN_PATH/bin:$PATH"
```

(same below) for consistency within this file.

Or even better: use 

```
echo "##vso[task.prependpath]$TOOLCHAIN_PATH/bin"
```
in the "Install toolchain" step to have PATH adjusted for all subsequent jobs.

(https://docs.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#prependpath-prepend-a-path-to-the--path-environment-variable)

**Review by imphil — APPROVED** [review]
LGTM with comments. Thanks!

File: azure-pipelines.yml (hunk 1 above)
Comment (rswarbrick) [reply azure-pipelines.yml:395]: Ah, neat, I'll do that. Thanks!

**Comment by rswarbrick** [comment]
Thanks for the review, Philipp. I've just re-pushed with a hacky "TMP" commit to make it trigger CI again and check that things won't break. If this goes through, I'll drop the TMP commit and merge.

**Pull request merged**